source/navitiacommon/navitiacommon/models.py

```diff
@@ -401,6 +401,19 @@ def get_from_id_or_name(cls, id=None, name=None):
         else:
             raise Exception({'error': 'instance is required'}, 400)
 
+    def delete_dataset(self):
+        jobs = db.session.query(Job).filter(Job.instance_id == self.id).all()
+        for j in jobs:
+            for dataset in j.data_sets.all():
+                if dataset.type == 'poi':
+                    db.session.query(Metric).filter(Metric.dataset_id == dataset.id).delete()
+                    db.session.delete(dataset)
+            db.session.commit()
```

Comment:
you don't have to commit after every query, you can (and must) commit only at the end

Reply:
Changed